src/sql/workbench/contrib/notebook/browser/notebookViews/notebookViewsActions.ts

```diff
@@ -0,0 +1,245 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Action } from 'vs/base/common/actions';
+import { ViewOptionsModal } from 'sql/workbench/contrib/notebook/browser/notebookViews/viewOptionsModal';
+import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { localize } from 'vs/nls';
+import { InsertCellsModal } from 'sql/workbench/contrib/notebook/browser/notebookViews/insertCellsModal';
+import { ComponentFactoryResolver, ViewContainerRef } from '@angular/core';
+import { IDialogService } from 'vs/platform/dialogs/common/dialogs';
+import { IDisposable } from 'vs/base/common/lifecycle';
+import { CellExecutionState, ICellModel, ViewMode } from 'sql/workbench/services/notebook/browser/models/modelInterfaces';
+import { CellActionBase, CellContext, IMultiStateData, MultiStateAction } from 'sql/workbench/contrib/notebook/browser/cellViews/codeActions';
+import { INotificationService, Severity } from 'vs/platform/notification/common/notification';
+import { IConnectionManagementService } from 'sql/platform/connection/common/connectionManagement';
+import { IKeybindingService } from 'vs/platform/keybinding/common/keybinding';
+import { ILogService } from 'vs/platform/log/common/log';
+import { getErrorMessage } from 'vs/base/common/errors';
+import * as types from 'vs/base/common/types';
+import { ActionBar, ActionsOrientation } from 'vs/base/browser/ui/actionbar/actionbar';
+import { Separator } from 'sql/base/browser/ui/separator/separator';
+import { ToggleMoreActions } from 'sql/workbench/contrib/notebook/browser/cellToolbarActions';
+import { NotebookViewsExtension } from 'sql/workbench/services/notebook/browser/notebookViews/notebookViewsExtension';
+import { INotebookView } from 'sql/workbench/services/notebook/browser/notebookViews/notebookViews';
+
+export class ViewSettingsAction extends Action {
+	private static readonly ID = 'notebookView.viewSettings';
+	private static readonly LABEL = undefined;
+	private static readonly ICON = 'notebook-button settings masked-icon';
+
+	constructor(
+		private _context: NotebookViewsExtension,
+		@IInstantiationService private _instantiationService: IInstantiationService,
+	) {
+		super(ViewSettingsAction.ID, ViewSettingsAction.LABEL, ViewSettingsAction.ICON);
+	}
+
+	override run(): Promise<void> {
+		return new Promise<void>((resolve) => {
+			const optionsModal = this._instantiationService.createInstance(ViewOptionsModal, this._context.getActiveView());
+			optionsModal.render();
+			optionsModal.open();
+			resolve();
```

Comment:
Same as the other one - this doesn't really need to be wrapped in a promise. You can just return Promise.resolve() or make it async